### PR TITLE
Add RabbitMQ workflows

### PR DESCRIPTION
**Pull request by nebilbenmabrouk**
- Remove a fake RabbitMQ workflow in the big-data bucket.
- Add the workflow RabbitMQ_Messaging to  the big-data bucket.
- Enhance the RabbitMQ PSA service.

### CODE REVIEW

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml [hunk 1]
```diff
@@ -0,0 +1,1321 @@
...
+    <variable advanced="false" description="Password to be used to access the Web UI of RabbitMQ. Its default value is 'activeeon'." group="Service parameters" hidden="false" model="PA:HIDDEN" name="PASSWORD" value="ENC(e5bZ1ZqClDcte6FNFHBE/to8QywjJMY3)"/>
+    <variable advanced="false" description="Exchange to be created in RabbitMQ in order to send messages" group="Messaging parameters" hidden="false" model="PA:NOT_EMPTY_STRING" name="EXCHANGE" value="rabbitmq-exchange"/>
+    <variable advanced="false" description="Queue to be created in RabbitMQ in order to send and receive messages" group="Messaging parameters" hidden="false" model="PA:NOT_EMPTY_STRING" name="QUEUE" value="rabbitmq-queue"/>
+  </variables>
+  <description>
+    <![CDATA[ A workflow that starts a RabbitMQ service (based on Docker), then uses the REST API of RabbitMQ to create an exchange and a queue given as input. The workflow further exposes user actions in order to send/read messages to/from RabbitMQ, respecively. Finally, when RabbitMQ is no longer needed, the user can terminate it via the action 'Stop_RabbitMQ'. ]]>
```
Comment (AlexFalessi) [inline RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:12]: Typo in respecively ? = respectively

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml [hunk 2]
```diff
@@ -0,0 +1,1321 @@
...
+</genericInformation>
+  <taskFlow>
+    <task fork="false" name="start_rabbitmq" onTaskError="cancelJob">
+      <description>
+        <![CDATA[ Start RabbitMQ as a service. ]]>
+      </description>
+      <variables>
+        <variable advanced="false" description="Name of the Tensorboard service instance to use." hidden="false" inherited="false" model="PA:NOT_EMPTY_STRING" name="RABBITMQ_INSTANCE_NAME" value="rabbitmq-${PA_JOB_ID}"/>
+        <variable advanced="false" description="The Tensorboard service activation workflow. Please keep the default value for this variable." hidden="false" inherited="false" model="PA:CATALOG_OBJECT(Workflow/psa,,,RabbitMQ%)" name="SERVICE_ACTIVATION_WORKFLOW" value="service-automation/RabbitMQ"/>
+        <variable advanced="false" description="Instance name of RabbitMQ" hidden="false" inherited="true" name="INSTANCE_NAME" value=""/>
```
Comment (AlexFalessi) [inline RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:27]: I don't understand the variables here.

"Name of the Tensorboard service instance to use" is named `RABBITMQ_INSTANCE_NAME` with value `rabbitmq-${PA_JOB_ID}`
And below "Instance name of RabbitMQ" is named `INSTANCE_NAME` and has empty value

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml [hunk 3]
```diff
@@ -0,0 +1,1321 @@
...
+    <task fork="true" name="wait_for_user_action">
+      <description>
+        <![CDATA[ Wait for a user action to: (i) send a message, (ii) read messages, or (iii) stop RabbitMQ. ]]>
+      </description>
+      <genericInformation>
+        <info name="task.icon" value="/automation-dashboard/styles/patterns/img/wf-icons/signal-wait.png"/>
+        <info name="task.documentation" value="user/ProActiveUserGuide.html#_branch"/>
+      </genericInformation>
+      <depends>
+        <task ref="cerate_binding"/>
```
Comment (AlexFalessi) [inline RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:78]: create_binding ?

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml [hunk 4]
```diff
@@ -0,0 +1,1321 @@
...
+        <script>
+          <code language="groovy">
+            <![CDATA[
+import org.ow2.proactive.scheduler.common.job.JobVariable
+
+def signals = ['Stop_RabbitMQ','Send_Message', 'Read_Messages']
+
+signalapi.readyForSignal("Stop_RabbitMQ")
+
+List <JobVariable> sendingMessageVariables = new java.util.ArrayList<JobVariable>()
```
Comment (AlexFalessi) [inline RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:90]: You don't need to specify the full package name `java.util.`

Also it looks like you can do a very little optimization doing

```
JobVariable jobVariables = new JobVariable("MESSAGE_TO_SEND", "put-here-your-message", "PA:NOT_EMPTY_STRING", "Message to send to RabbitMQ", "Message sending", false, false))
signalapi.readyForSignal("Send_Message", Arrays.asList(jobVariables)); 
```

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml [hunk 5]
```diff
@@ -0,0 +1,1321 @@
...
+import io.restassured.http.ContentType;
+import io.restassured.RestAssured;
+import com.google.common.base.Strings;
+import groovy.json.JsonSlurper;
+import io.restassured.specification.ProxySpecification
+
+debug = Boolean.parseBoolean(variables.get("DEBUG"))
+
+if(variables.get("PROXY_HOST")?.trim() && variables.get("PROXY_PORT")?.trim()){
+  proxySpecification = new ProxySpecification(variables.get("PROXY_HOST"), variables.get("PROXY_PORT"), variables.get("PROXY_SCHEME"))
```
Comment (AlexFalessi) [inline RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:375]: No need to check `variables.get("PROXY_SCHEME")` ?

File: RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml (hunk 3 above)
Comment (nebilbenmabrouk) [reply RabbitMQ/resources/catalog/RabbitMQ_Messaging.xml:78]: Done.

**Pull request merged**